reark/src/main/java/io/reark/reark/data/store/ContentProviderStore.java

```diff
@@ -76,44 +75,50 @@ protected ContentProviderStore(@NonNull ContentResolver contentResolver) {
 
         updateSubject
                 .onBackpressureBuffer()
-                .observeOn(Schedulers.io())
-                .subscribe(pair -> {
-                    updateIfValueChanged(this, pair);
-                });
+                .doOnNext(pair -> locker.createLock(pair.second))
+                .flatMap(pair -> updateIfValueChanged(this, pair.first, pair.second))
+                .subscribe(locker::freeLock, error -> Log.e(TAG, "Error updating", error));
```

Comment:
Does it mean that if there is an error in updateIfValueChanged the lock is never freed for the value?